New view for branches in Iceberg repositories

AlexisCnockaert opened the pull request:
- Display branches as a tree to have a better look, similar as Git Kraken or the ```tree``` command from Linux
Fixes: #1922 .
![image](https://github.com/user-attachments/assets/082c001d-0eba-4666-9313-4f7b381f5f2b)

AlexisCnockaert committed:
New view for branches in Iceberg repositories

- Display branches as a tree to have a better look, similar as Git Kraken or the ```tree``` command from Linux

jecisc reviewed (changes requested):
- It would be nice to add an #expandAll 
- It would be nice to sort the items 
- The children need to be recursive `baseline/toto/tata`
- Disable OK button if we do not select a branch but a category

AlexisCnockaert committed:
addition of ```expandAll``` for branches

AlexisCnockaert committed:
Ok button inform the user if the selected item is a group and not a branch

AlexisCnockaert committed:
Branches now uses recursivity on its children to be displayed

AlexisCnockaert commented:
Now looks like 
![image](https://github.com/user-attachments/assets/7d982a7c-5486-475e-83c8-589723519869)

AlexisCnockaert committed:
change the branchList into a ```SpTreePresenter```

astares commented:
@AlexisCnockaert Can we add a (search) filter too (if the number of branches is high)

AlexisCnockaert committed:
Ugly code but everything works at this point

AlexisCnockaert commented:
@astares yes sure, no problem

AlexisCnockaert committed:
remove useless code

AlexisCnockaert committed:
not sure

AlexisCnockaert committed:
commit finish

Pharo13.0.0:
#Branch: modif Fixes #
#Please use one of those
#feat:
#fix:
#refactor:
#chore:
#comment:

AlexisCnockaert committed:
Created a class to represent branch items and simplified the whole code

jecisc commented:
> @AlexisCnockaert Can we add a (search) filter too (if the number of branches is high)

In fact we enabled it but it does not appear. This seems to be a Spec bug

demarey commented:
Looks like there is a problem with this PR: I got a DNU when fixing a repo by checking out a branch.

<img width="1712" alt="Capture d’écran 2025-05-06 à 16 19 00" src="https://github.com/user-attachments/assets/45575c8c-3fb6-4b05-ba8e-334a7c9be63c" />

AlexisCnockaert commented:
Yes this PR need to make changes to methods calling the selectedBranch, I will fix this

AlexisCnockaert commented:
Fixed in #1931
